src/Http/Controllers/Support/ResolveController.php

```diff
@@ -78,48 +80,143 @@ public function resolveIdsToNames(Request $request)
 
         })->chunk(1000)->each(function ($chunk) use (&$response, $eseye) {
 
-            // universe resolver is not working on factions at this time
-            // retrieve them from SDE and remove them from collection
-            // TODO CCP WIP : https://github.com/ccpgames/esi-issues/issues/736
-            $names = ChrFaction::whereIn('factionID', $chunk->flatten()->toArray())
-                ->get();
+            $chunk = $this->resolveFactionIDs($chunk, $response);
 
-            collect($names)->each(function ($name) use (&$response) {
+            // quick break if no more IDs must be resolve by ESI
+            if ($chunk->count() == 0)
+                return;
 
-                cache([$this->prefix . $name->factionID => $name->factionName], carbon()->addCentury());
-                $response[$name->factionID] = $name->factionName;
+            $chunk = $this->resolveCharacterIDsFromSeat($chunk, $response);
 
-            });
+            // quick break if no more IDs must be resolve by ESI
+            if ($chunk->count() == 0)
+                return;
 
-            $chunk = $chunk->filter(function ($id) use ($names) {
-
-                return ! $names->contains('factionID', $id);
-            });
+            $chunk = $this->resolveCorporationIDsFromSeat($chunk, $response);
 
             // quick break if no more IDs must be resolve by ESI
             if ($chunk->count() == 0)
```

Comment:
rather than counting (which we don't really care about here), prefer isEmpty instruction : https://laravel.com/docs/5.6/collections#method-isempty